arrows/core/derive_metadata.cxx

```diff
@@ -210,47 +237,86 @@ double
 compute_gsd( kwiver::vital::metadata_sptr const& metadata,
              size_t frame_width, size_t frame_height )
 {
-  if ( frame_width == 0 || frame_height == 0 )
+  if( frame_width < 1 || frame_height < 1 )
   {
-    VITAL_THROW( kv::invalid_value, "frame dimensions cannot be zero" );
+    VITAL_THROW( kv::invalid_value, "frame dimensions must both be positive" );
   }
 
-  kv::vector_2d sensor_fov;
+  // RP1201 method
   try
   {
-    // Attempt to acquire sensor FOV
-    sensor_fov = get_sensor_fov( metadata );
+    // Intermediate values
+    kv::rotation_d const total_rotation = get_total_rotation( metadata );
+    double yaw, pitch, roll;
+    total_rotation.get_yaw_pitch_roll( yaw, pitch, roll );
+    // Ideally slant range should be pre-calculated
+    double const slant_range = get_slant_range( metadata );
+    double const sensor_horizontal_fov_rad = get_sensor_horizontal_fov( metadata );
+    double const sensor_vertical_fov_rad = get_sensor_vertical_fov( metadata );
+
+    if( slant_range > 0 && slant_range <= 5000000 &&
+        sensor_horizontal_fov_rad >= 0 &&
+        sensor_horizontal_fov_rad <= kv::pi &&
+        sensor_vertical_fov_rad >= 0 && sensor_vertical_fov_rad <= kv::pi )
```

Comment:
I would add a line break here to make the formatting consistent with the horizontal case above.